src/contract.py

```diff
@@ -1,21 +1,34 @@
 from __future__ import annotations
 from abc import ABC, abstractmethod
+
+import numpy as np
+
 from src.enums import *
+from statistics import mean
 
 
 class Contract(ABC):
-    def __init__(self, und: Stock, dtype: PutCallFwd, strk: float, exp: float) -> None:
+    def __init__(self, und: Stock, dtype: PutCallFwd, longshort: LongShort, strk: float, exp: float, num_mon: int = 1) -> None:
         self._underlying: Stock = und
         self._derivative_type: PutCallFwd = dtype
+        self._derivative_longshort: LongShort = longshort
+        self.ls = 1 if longshort == LongShort.LONG else -1
```

Comment:
Let's have this as `_ls` to indicate that no one should modify it outside of the class. It's a non-public property.

Reply:
done